lib/cereal/postgres/task_cleaner.go

```diff
@@ -0,0 +1,95 @@
+package postgres
+
+import (
+	"context"
+	"database/sql"
+	"math"
+	"sync"
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+type taskCleaner struct {
+	db      *sql.DB
+	wgStart sync.WaitGroup
+	wgStop  sync.WaitGroup
+	stop    context.CancelFunc
+
+	checkInterval time.Duration
+	taskTimeout   time.Duration
+}
+
+func newTaskCleaner(db *sql.DB) *taskCleaner {
+	cleaner := &taskCleaner{
+		db:            db,
+		checkInterval: 60 * time.Second,
+		taskTimeout:   300 * time.Second,
+	}
+	cleaner.wgStart.Add(1)
+	return cleaner
+}
+
+func (w *taskCleaner) Start(ctx context.Context) {
+	// Make sure this function is only called once. A second call will
+	// cause this to panic
+	w.wgStart.Done()
```

Comment:
💭 Turning this into a panic is a nice defense against what we see as a programming mistake, provided we have sufficient test coverage.  I suppose an alternative would be to use sync.Once to make subsequence calls to Start() meaningless.